src/components/pages/PeoplePage/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,201 @@
+import classNames from 'classnames';
+import { useCallback, useEffect, useState } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { PersonLink } from '../PersonLink';
+import { Person } from '../../../../types/Person';
+import { SearchLink } from '../SearchLink';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { personSlug } = useParams();
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>(people);
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sortBy = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const findParent = useCallback((personName: string | null) => {
+    const parent = people.find(person => person.name === personName);
+
+    if (parent) {
+      return (
+        <PersonLink person={parent} />
+      );
+    }
+
+    return personName || '-';
+  }, [people]);
+
+  useEffect(() => {
+    const filteredAndSortedPeople = people.filter(person => {
```

Comment:
Smart chaining, but looks terrible. Cut it on parts - it much more easy to manage in future. Believe me.